### PR TITLE
Capturinginstallreferrer

### PR DESCRIPTION
**Pull request by EvangelosG**
this PR adds the string from the play store's install broadcast (named referrer) to the v1/open call as install_referrer.

Things to do:

1. set link click id with this value, do not include as install_referrer
2. grab data in onIntentReady vs onInstallReferrerFetchFinished interface? reducing lock count but potentially increasing complexity? (although adding locks inherently increases complexity). so far it has proved unreliable to use onIntentReady.
3. figure out if we can create install listener programmatically w/o manifest entry.

**test using the following:**

Launch app from Android Studio, it will freeze (wait) for 5 seconds, you can unlock by issuing the following:

adb shell am broadcast -a com.android.vending.INSTALL_REFERRER -n io.branch.branchandroiddemo/io.branch.referral.InstallListener --es "referrer" "link_click_id=123555232345"

enable via: Branch.enableMatchGuaranteed() in app class before getAutoInstance.
set timeout via setReferralFetchWaitTime for how long app should wait for play store's broadcast before continuing.

### CODE REVIEW

**Comment by EvangelosG** [comment]
<s>1. set link click id with this value, do not include as install_referrer</s>

File: Branch-SDK-TestBed/src/io/branch/branchandroiddemo/CustomBranchApp.java [hunk 1]
```diff
@@ -3,12 +3,18 @@
 
 //import com.squareup.leakcanary.LeakCanary;
 
+import android.app.Application;
+
+import io.branch.referral.Branch;
 import io.branch.referral.BranchApp;
 
-public final class CustomBranchApp extends BranchApp {
+public final class CustomBranchApp extends Application {
```
Comment (sojanpr) [inline Branch-SDK-TestBed/src/io/branch/branchandroiddemo/CustomBranchApp.java]: Any specific reason for not extending BranchApp?

Comment (EvangelosG) [inline Branch-SDK-TestBed/src/io/branch/branchandroiddemo/CustomBranchApp.java]: only for testing so i can make calls before getAutoInstance

Comment (EvangelosG) [inline Branch-SDK-TestBed/src/io/branch/branchandroiddemo/CustomBranchApp.java]: tested w/ BranchApp, works as well! :)

**Comment by sojanpr** [comment]
LGTM 👍  (hope it is tested extensively)

**Pull request merged**